test/test_images/probe_helper/probe_helper.go

```diff
@@ -87,51 +106,141 @@ func forwardFromProbe(ctx context.Context, brokerClient cloudevents.Client, pubs
 		defer cancel()
 		switch event.Type() {
 		case BrokerE2EDeliveryProbeEventType:
-			receiverChannel, err = receivedEvents.createReceiverChannel(event)
+			channelID = event.ID()
+			receiverChannel, err = receivedEvents.createReceiverChannel(channelID)
 			if err != nil {
-				return cloudevents.NewReceipt(false, "Probe forwarding failed, could not create receiver channel: %v", err)
+				log.Printf("Probe forwarding failed, could not create receiver channel: %v", err)
+				return cloudevents.ResultNACK
 			}
+			defer receivedEvents.deleteReceiverChannel(channelID)
+
 			// The broker client forwards the event to the broker.
 			if res := brokerClient.Send(ctx, event); !cloudevents.IsACK(res) {
 				log.Printf("Error when sending event %v to broker: %+v \n", event.ID(), res)
 				return res
 			}
 		case CloudPubSubSourceProbeEventType:
-			receiverChannel, err = receivedEvents.createReceiverChannel(event)
+			channelID = event.ID()
+			receiverChannel, err = receivedEvents.createReceiverChannel(channelID)
 			if err != nil {
-				return cloudevents.NewReceipt(false, "Probe forwarding failed, could not create receiver channel: %v", err)
+				log.Printf("Probe forwarding failed, could not create receiver channel: %v", err)
+				return cloudevents.ResultNACK
 			}
+			defer receivedEvents.deleteReceiverChannel(channelID)
+
 			// The pubsub client forwards the event as a message to a pubsub topic.
 			if res := pubsubClient.Send(ctx, event); !cloudevents.IsACK(res) {
 				log.Printf("Error when publishing event %v to pubsub topic: %+v \n", event.ID(), res)
 				return res
 			}
+		case CloudStorageSourceProbeEventType:
+			obj := bucket.Object(event.ID())
```

Comment:
If I understand this correctly, we assume all four different types of CloudStorage probes are executed with the same event ID in a certain order, right? If so, please document it.

Reply:
Documented the probes in [main.go](https://github.com/google/knative-gcp/pull/1639/files#diff-502ab32f599d1dd7e41dd1c389b6bf5cR16-R60)